stock_available_to_promise_release/models/stock_move.py

```diff
@@ -25,8 +25,32 @@ class StockMove(models.Model):
         help="Available to Promise quantity minus quantities promised "
         " to older promised operations.",
     )
+    release_ready = fields.Boolean(
+        compute="_compute_release_ready", search="_search_release_ready",
+    )
     need_release = fields.Boolean(index=True,)
 
+    @api.depends(
+        "ordered_available_to_promise", "picking_id.move_type", "picking_id.move_lines"
+    )
+    def _compute_release_ready(self):
+        for move in self:
+            if move.picking_id.move_type == "one":
+                move.release_ready = all(
+                    m.ordered_available_to_promise > 0
+                    for m in move.picking_id.move_lines
+                )
+            else:
+                move.release_ready = move.ordered_available_to_promise > 0
+
+    @api.model
+    def _search_release_ready(self, operator, value):
+        if operator != "=":
+            raise UserError(_("Unsupported operator %s") % (operator,))
+        moves = self.search([("ordered_available_to_promise", ">", 0)])
```

Comment:
Maybe rather than relying on the search method of `ordered_available_to_promise` (which browse *all* the moves), and then filtering on `release_ready`, which need to iterate over the values of `ordered_available_to_promise` again (although it should be cached this time), it could be faster to do it the other way around:

* search the stock.picking that need release (would need a new search method on the field...)
* for each picking, if the move_type is "one", check if all `ordered_available_to_promise > 0`, otherwise, check if at least one is `ordered_available_to_promise > 0`

We would need to measure the improvement, can be done later.

Reply:
Makes sense, it's is more efficient this way.